### PR TITLE
Upgrade ciris, cats-core, cats-effect, awssdk, scala and sbt versions

### PR DESCRIPTION
**Pull request by alexgb1**
Initially I just wanted to upgrade ciris from 2.x to 3.x, but ended up upgrading the other dependencies as well. This might require a minor or major release because ciris 3.x is incompatible with 2.x.

### CODE REVIEW

**Comment by alexgb1** [comment]
Getting : 
```
[error] Caused by: dotty.tools.dotc.core.CyclicReference: 
[error]         at dotty.tools.dotc.core.CyclicReference$.apply(TypeErrors.scala:159)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:171)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$innerClasses$.getMember$1(ClassfileParser.scala:1097)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$innerClasses$.classSymbol(ClassfileParser.scala:1118)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.classNameToSymbol(ClassfileParser.scala:116)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.sig2type$1(ClassfileParser.scala:434)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.processTypeArgs$1(ClassfileParser.scala:421)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.sig2type$1(ClassfileParser.scala:436)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.dotty$tools$dotc$core$classfile$ClassfileParser$$sigToType(ClassfileParser.scala:560)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$AttributeCompleter.complete(ClassfileParser.scala:703)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.parseClass(ClassfileParser.scala:219)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.$anonfun$1(ClassfileParser.scala:87)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.run(ClassfileParser.scala:82)
[error]         at dotty.tools.dotc.core.ClassfileLoader.load(SymbolLoaders.scala:411)
[error]         at dotty.tools.dotc.core.ClassfileLoader.doComplete(SymbolLoaders.scala:406)
[error]         at dotty.tools.dotc.core.SymbolLoader.complete(SymbolLoaders.scala:340)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:173)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeOnce(SymDenotations.scala:372)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.unforcedDecls(SymDenotations.scala:396)
[error]         at dotty.tools.dotc.core.SymDenotations$ClassDenotation.typeParamsFromDecls(SymDenotations.scala:1803)
[error]         at dotty.tools.dotc.core.SymDenotations$ClassDenotation.typeParams(SymDenotations.scala:1816)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.apply(ClassfileParser.scala:38)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.apply(ClassfileParser.scala:37)
[error]         at scala.collection.immutable.List.mapConserve(List.scala:472)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.apply(ClassfileParser.scala:42)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.apply(ClassfileParser.scala:37)
[error]         at scala.collection.immutable.List.mapConserve(List.scala:472)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$$anon$2.apply(ClassfileParser.scala:47)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser$AttributeCompleter.complete(ClassfileParser.scala:729)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.parseClass(ClassfileParser.scala:219)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.$anonfun$1(ClassfileParser.scala:87)
[error]         at dotty.tools.dotc.core.classfile.ClassfileParser.run(ClassfileParser.scala:82)
[error]         at dotty.tools.dotc.core.ClassfileLoader.load(SymbolLoaders.scala:411)
[error]         at dotty.tools.dotc.core.ClassfileLoader.doComplete(SymbolLoaders.scala:406)
[error]         at dotty.tools.dotc.core.SymbolLoader$$anon$1.doComplete(SymbolLoaders.scala:325)
[error]         at dotty.tools.dotc.core.SymbolLoader.complete(SymbolLoaders.scala:340)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:173)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:380)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.flags(SymDenotations.scala:66)
[error]         at dotty.tools.dotc.core.SymDenotations$ModuleCompleter.complete(SymDenotations.scala:2706)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.completeFrom(SymDenotations.scala:173)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.completeInfo$1(Denotations.scala:187)
[error]         at dotty.tools.dotc.core.Denotations$Denotation.info(Denotations.scala:189)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.ensureCompleted(SymDenotations.scala:380)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.flags(SymDenotations.scala:66)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.is(SymDenotations.scala:112)
[error]         at dotty.tools.dotc.core.SymDenotations$SymDenotation.isStableMember(SymDenotations.scala:752)
[error]         at dotty.tools.dotc.core.Types$Type.widenIfUnstable(Types.scala:1261)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTerm$1(TreeUnpickler.scala:1301)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTerm(TreeUnpickler.scala:1394)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTerm$1(TreeUnpickler.scala:1233)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTerm(TreeUnpickler.scala:1394)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.$anonfun$24(TreeUnpickler.scala:1234)
[error]         at dotty.tools.tasty.TastyReader.until(TastyReader.scala:125)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTerm$1(TreeUnpickler.scala:1234)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTerm(TreeUnpickler.scala:1394)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.$anonfun$24(TreeUnpickler.scala:1234)
[error]         at dotty.tools.tasty.TastyReader.until(TastyReader.scala:125)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTerm$1(TreeUnpickler.scala:1234)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTerm(TreeUnpickler.scala:1394)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readLengthTerm$1(TreeUnpickler.scala:1233)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$TreeReader.readTerm(TreeUnpickler.scala:1394)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler.readRhs$1$$anonfun$1$$anonfun$1(TreeUnpickler.scala:858)
[error]         at dotty.tools.dotc.core.tasty.TreeUnpickler$LazyReader.complete(TreeUnpickler.scala:1531)
[error]         at dotty.tools.dotc.ast.Trees$WithLazyField.forceIfLazy(Trees.scala:1023)
[error]         at dotty.tools.dotc.ast.Trees$WithLazyField.forceIfLazy$(Trees.scala:1018)
[error]         at dotty.tools.dotc.ast.Trees$ValOrDefDef.forceIfLazy(Trees.scala:412)
[error]         at dotty.tools.dotc.ast.Trees$ValOrDefDef.rhs(Trees.scala:417)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1617)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1658)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:163)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1529)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1531)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1624)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1658)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:163)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1621)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1658)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:163)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1529)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1531)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1630)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1658)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:163)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.apply(Trees.scala:1657)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.fold$1(Trees.scala:1529)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.apply(Trees.scala:1531)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1630)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeAccumulator.foldOver(Trees.scala:1535)
[error]         at dotty.tools.dotc.ast.Trees$Instance$TreeTraverser.traverseChildren(Trees.scala:1658)
[error]         at dotty.tools.dotc.CompilationUnit$Force.traverse(CompilationUnit.scala:163)
[error]         at dotty.tools.dotc.CompilationUnit$.apply(CompilationUnit.scala:119)
[error]         at dotty.tools.dotc.CompilationUnit$.apply(CompilationUnit.scala:110)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.compilationUnit$1(ReadTasty.scala:42)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.readTASTY(ReadTasty.scala:70)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.runOn$$anonfun$1(ReadTasty.scala:25)
[error]         at scala.collection.immutable.List.flatMap(List.scala:293)
[error]         at dotty.tools.dotc.fromtasty.ReadTasty.runOn(ReadTasty.scala:25)
[error]         at dotty.tools.dotc.Run.runPhases$1$$anonfun$1(Run.scala:238)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1321)
[error]         at dotty.tools.dotc.Run.runPhases$1(Run.scala:249)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:257)
[error]         at dotty.tools.dotc.Run.compileUnits$$anonfun$adapted$1(Run.scala:266)
[error]         at dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:68)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:266)
[error]         at dotty.tools.dotc.Run.compileUnits(Run.scala:196)
[error]         at dotty.tools.dotc.fromtasty.TASTYRun.compile(TASTYRun.scala:14)
[error]         at dotty.tools.dotc.Driver.doCompile(Driver.scala:35)
[error]         at dotty.tools.dotc.Driver.process(Driver.scala:195)
[error]         at scala.tasty.inspector.OldTastyInspector.inspectFilesInContext(OldTastyInspector.scala:72)
[error]         at scala.tasty.inspector.OldTastyInspector.inspectFilesInContext$(OldTastyInspector.scala:22)
[error]         at scala.tasty.inspector.DocTastyInspector.inspectFilesInContext(DocTastyInspector.scala:5)
[error]         at dotty.tools.scaladoc.tasty.ScaladocTastyInspector.result(TastyParser.scala:147)
[error]         at dotty.tools.scaladoc.ScalaModuleProvider$.mkModule(ScalaModuleProvider.scala:11)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:241)
[error]         at dotty.tools.scaladoc.Scaladoc$.run$$anonfun$1(Scaladoc.scala:73)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error]         at scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error]         at scala.Option.map(Option.scala:242)
[error]         at dotty.tools.scaladoc.Scaladoc$.run(Scaladoc.scala:77)
[error]         at dotty.tools.dottydoc.Main$.process(Main.scala:25)
[error]         at dotty.tools.dottydoc.Main.process(Main.scala)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error]         at xsbt.DottydocRunner.run(DottydocRunner.java:61)
[error]         at xsbt.ScaladocInterface.run(ScaladocInterface.java:11)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[error]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[error]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[error]         at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[error]         at sbt.internal.inc.AnalyzingCompiler.invoke(AnalyzingCompiler.scala:329)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:175)
[error]         at sbt.internal.inc.AnalyzingCompiler.doc(AnalyzingCompiler.scala:133)
[error]         at sbt.Doc$.$anonfun$scaladoc$1(Doc.scala:52)
[error]         at sbt.Doc$.$anonfun$scaladoc$1$adapted(Doc.scala:40)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1(RawCompileLike.scala:79)
[error]         at sbt.RawCompileLike$.$anonfun$prepare$1$adapted(RawCompileLike.scala:72)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4(RawCompileLike.scala:63)
[error]         at sbt.RawCompileLike$.$anonfun$cached$4$adapted(RawCompileLike.scala:61)
[error]         at sbt.util.Tracked$.$anonfun$inputChangedW$1(Tracked.scala:219)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1(RawCompileLike.scala:68)
[error]         at sbt.RawCompileLike$.$anonfun$cached$1$adapted(RawCompileLike.scala:52)
[error]         at sbt.Defaults$.$anonfun$docTaskSettings$4(Defaults.scala:2156)
[error]         at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[error]         at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[error]         at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[error]         at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[error]         at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[error]         at sbt.Execute.work(Execute.scala:291)
[error]         at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[error]         at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[error]         at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[error]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[error]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[error]         at java.base/java.lang.Thread.run(Thread.java:829)
[error] (Compile / doc) java.lang.reflect.InvocationTargetException

```

when trying to publish locally, the only fix I found is `Compile / doc / sources := Nil`

**Comment by alexgb1** [comment]
There is a fix for it, https://github.com/lampepfl/dotty/pull/16882, not yet released

File: build.sbt [hunk 1]
```diff
@@ -2,15 +2,15 @@ name := "ciris-aws-secretsmanager"
 organization := "com.ovoenergy"
 licenses += ("MIT", url("http://opensource.org/licenses/MIT"))
 
-scalaVersion := "3.1.0"
-crossScalaVersions := Seq(scalaVersion.value, "2.13.5", "2.12.13")
+scalaVersion := "3.2.2"
```
Comment (keirlawson) [inline build.sbt:5]: Is there a reason to drop scala 3.1 support?

File: build.sbt [hunk 2]
```diff
@@ -47,4 +47,6 @@ releaseProcess := Seq[ReleaseStep](
   setNextVersion,
   commitNextVersion,
   pushChanges
-)
\ No newline at end of file
+)
+
+Compile / doc / sources := Nil
```
Comment (keirlawson) [inline build.sbt:52]: What's the purpose of this?

File: build.sbt (hunk 1 above)
Comment (alexgb1) [reply build.sbt:5]: Not really, happy to revert this bit.

File: build.sbt (hunk 2 above)
Comment (alexgb1) [reply build.sbt:52]: See https://github.com/lampepfl/dotty/issues/15288#issuecomment-1324738643

File: build.sbt (hunk 1 above)
Comment (alexgb1) [reply build.sbt:5]: Also, maybe I'm confused, but I was thinking of this bit of scala docs:
```
Thanks to the guarantees of Scala 3, there is no longer a need for cross-compilation. For example, the migration from Scala 3.1 to Scala 3.2 for the large majority of the projects would only require changing the version in the build definition
```

Comment (alexgb1) [reply build.sbt:5]: from https://www.scala-lang.org/blog/2022/08/17/long-term-compatibility-plans.html#:~:text=Thanks%20to%20the%20guarantees%20of,version%20in%20the%20build%20definition.

Comment (alexgb1) [reply build.sbt:5]: And the latest ciris depends on scala 3.2 https://github.com/vlovgr/ciris/releases/tag/v3.1.0 , therefore we need to upgrade this as well...

**Review by keirlawson — APPROVED** [review]

**Pull request merged**